superset-frontend/plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -226,36 +226,61 @@ const transformProps = (
     emitCrossFilters,
   } = chartProps;
 
+  const [metrics, percentMetrics, columns] = processColumns(chartProps);
+
+  let baseQuery;
+  let countQuery;
+  let totalQuery;
+  let rowCount;
+  if (formData.server_pagination) {
+    [baseQuery, countQuery, totalQuery] = queriesData;
+    rowCount = (countQuery?.data?.[0]?.rowcount as number) ?? 0;
+  } else {
+    [baseQuery, totalQuery] = queriesData;
+    rowCount = baseQuery?.rowcount ?? 0;
+  }
+  const data = processDataRecords(baseQuery?.data, columns);
+
+  if (Number(formData.server_page_length) > 10000) {
+    formData.server_page_length = 10000;
+  }
+
+  if (
+    formData.server_pagination &&
+    !Number.isNaN(formData.server_page_length) &&
+    (formData.server_page_length || 0) <= data.length &&
+    !formData.server_page_length_options?.some(
+      (option: [number, string]) => option[0] === formData.server_page_length,
+    )
+  ) {
+    formData.server_page_length_options?.push([
+      Number(formData.server_page_length),
+      formData.server_page_length === 0
+        ? 'All'
```

Comment:
I think this should be localized

Reply:
I'm not sure what do you mean but I'm using the t function provided by @superset-ui/core as `t('All')` correct me if I'm wrong